verifier/src/tcb.rs

```diff
@@ -116,17 +137,58 @@ pub struct TcbComponent {
 /// <https://api.trustedservices.intel.com/sgx/certification/v4/tcb?fmspc={}>
 ///
 /// Due to the way the TCB info is signed the contents should be provided as is.
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Deserialize, Clone)]
 #[serde(rename_all = "camelCase")]
-pub struct TcbInfoRaw<'a> {
-    tcb_info: &'a RawValue,
-    signature: &'a str,
+pub struct TcbInfoRaw {
+    tcb_info: Box<RawValue>,
+    #[serde(with = "hex")]
+    signature: Vec<u8>,
+}
+
+impl TcbInfoRaw {
+    /// Verify the `tcbInfo` signature and time are valid.
+    ///
+    /// # Arguments
+    /// - `key` - The public key to verify the `tcbInfo` signature with
+    /// - `time` - The current system time
+    ///   This is expected to be generated by:
+    ///     ```ignore
+    ///     let time = DateTime::from_system_time(SystemTime::now()).unwrap();
+    ///     ```
+    ///   or equivalent
+    pub fn verify(self, key: &VerifyingKey, time: DateTime) -> Result<(), Error> {
```

Comment:
Up for suggestions on this API.
When doing the x509 I followed the pattern of `UnverfiedFoo.verify() -> VerifiedFoo` and then left most of the behavior on the verified version to kind of nudge clients to verify before looking at anything.  I kind of like the idea, not sure how it is in practice.  Also depends how we integerate with the verifier.